Wind-River/cloud-platform-deployment-manager
# Update factory install config workflow

Update factory install config workflow
As the factory install workflow changes: starting from fully deployed
system to only the active controller was deployed. This commit updates
the configuration workflow to align with the starting point: using the
day-1 operation rather than day-2 operation to config the system.

This commit:
1. Removes the strategy-related code for configuration post-factory
install.
2. Update all the resources' reconciled status to "false" once to
unblock the reconciliation.
3. Set the configuration post factory install finalized post the
unlock of the active controller.

With this change, the configuration post-factory install should no
longer trigger a vim strategy to orchestrate the lock/unlock of the
hosts.

Additional requests:
1 the host should be locked before applying the configMap to start the
configuration.
2 the administrative status should be set to "unlock" to trigger the
unlock of the host.

Test plan:
1. Deploy an active controller of a DX system, and lock the controller.
2. Apply the configMap to trigger the configuration.
3. Verify the host can be reconfigured and unlocked by DM.
4. Verify the host can be reconciled post-unlock.

Signed-off-by: Yuxing Jiang <Yuxing.Jiang@windriver.com>
